app/services/memberships/revoke_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Memberships
+  class RevokeService < BaseService
+    def revoke(id)
+      membership = Membership.find_by(id: id)
+      return result.fail!(code: 'not_found') unless membership
+
+      if result.user.id == membership.user.id
+        return result.fail!(
+          code: 'unprocessable_entity',
+          message: 'revoke_own_membership',
+        )
+      end
+
+      membership.mark_as_revoked!
+
+      result.membership = membership
+      result
+    rescue ActiveRecord::RecordInvalid => e
```

Comment:
How is it possible to have an `ActiveRecord::RecordInvalid` error?

Reply:
Yep makes no sense, removed